torch/distributed/distributed_c10d.py

```diff
@@ -2139,17 +2139,28 @@ def all_gather_coalesced(
     ]
 
     if group is None:
-        default_pg = _get_default_group()
-        work = default_pg.allgather_coalesced(output_tensor_lists, input_tensor_list)
-    else:
-        work = group.allgather_coalesced(output_tensor_lists, input_tensor_list)
-
+        group = _get_default_group()
+
+    flatten_tensors = [x.flatten().tolist() for x in input_tensor_list]
+    one_list = [x for i in flatten_tensors for x in i]
+    one_tensor = torch.Tensor(one_list)
+    temp_out = [torch.zeros(one_tensor.shape) for _ in range(len(output_tensor_lists))]
+
+    def copy_values_to_output_tensor_lists(fut_param):
+        vals = fut_param.wait()
+        sizes = list(map(lambda x: x.numel(), input_tensor_list))
+        for idx, val in enumerate(vals):
+            chunks = torch.split(val, sizes)
+            for i in range(len(output_tensor_lists[idx])):
+                output_tensor_lists[idx][i].copy_(chunks[i].reshape_as(input_tensor_list[i]))
+
+    work = group.allgather([temp_out], [one_tensor])
```

Comment:
Shall we update the var names to make the more explicit? They are coalesced outputs and coalesced input, and the name should be self-explaining.